### ISSUE TITLE
Player fog effect

### ISSUE THREAD

**Issue by tacotexmex** [minetest/minetest#7284]
##### Issue type
 - Feature request

##### Summary
Drawing on ideas from #6003 and #7035 it seems that a general per-player fog feature is beneficial.

It will be a great way to increase immersion by enabling different fog thickness and color for biomes, water and other liquids, weather effects, player effects (nearsightedness, farsightedness).

My speculation is that bendeutsch has already laid the ground work for this code-wise in #6003 and that it more or less needs to be applied to this new feature, correct me if I'm wrong.

**Comment by paramat** [comment]
Good idea. Currently fog distance is determined by view range, so view range will have to be overriden, or maybe better fog distance is set to the smaller of the set value and view range.

**Comment by TMcSquared** [comment]
or maybe a function creates a balance using those values and then caches it.

**Comment by tacotexmex** [comment]
Something for 5.1.0?
